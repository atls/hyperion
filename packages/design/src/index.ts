export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
