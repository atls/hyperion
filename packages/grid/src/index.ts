export * from './Column'
export * from './Grid'
