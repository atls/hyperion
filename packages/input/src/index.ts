export * from './Textarea'
export * from './Input'
export * from './Checkbox'
