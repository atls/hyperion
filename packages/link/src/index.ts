export * from './Link'
export * from './NextLink'
export * from './RouteLink'
