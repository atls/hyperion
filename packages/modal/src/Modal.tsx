import ScrollLock             from 'react-scrolllock'
import document               from 'global/document'
import React, { useRef }      from 'react'
import { motion }             from 'framer-motion'
import { createPortal }       from 'react-dom'
import { ifProp, switchProp } from 'styled-tools'

import styled                 from '@emotion/styled'

interface ContainerProps {
  visible: boolean
  opacity: string
}

const variants = {
  visible: { opacity: 1 },
  hidden: { opacity: 0 },
}

const isVisible = ifProp('visible', { display: 'flex', zIndex: 10 })

const opacities = switchProp('opacity', () => ({
  small: {
    background: 'rgba(0, 0, 0, 0.2)',
  },
  large: {
    background: 'rgba(0, 0, 0, 0.8)',
  },
}))

const StyledContainer = styled(motion.nav)<ContainerProps>(
  {
    width: '100%',
    height: '100%',
    position: 'fixed',
    display: 'none',
    background: 'rgba(0, 0, 0, 0.8)',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    overflowY: 'scroll',
    justifyContent: 'center',
    alignItems: 'center',
  },
  isVisible,
  opacities
)

export const Modal = ({ children, visible, onClose, opacity = 'large' }) => {
  if (typeof window !== 'undefined') {
    const node = useRef(null)

    const handleClick = event => {
      if (!(node.current && node.current.children[0].contains(event.target))) {
        onClose()
      }
    }

    if (visible) {
      return createPortal(
        <ScrollLock>
          <StyledContainer
            onClick={handleClick}
            animate={visible ? 'visible' : 'hidden'}
            variants={variants}
            visible={visible}
            opacity={opacity}
            ref={node}
          >
            {children}
          </StyledContainer>
        </ScrollLock>,
        document.body
      )
    }
  }
  return null
}
