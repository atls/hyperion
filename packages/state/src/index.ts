export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
