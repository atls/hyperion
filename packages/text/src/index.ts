export * from './Text'
export * from './Space'
export * from './SearchResult'
