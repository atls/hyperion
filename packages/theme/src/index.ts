import * as theme from './theme'

export * from './Provider'
export * from './globals'

export { theme }
