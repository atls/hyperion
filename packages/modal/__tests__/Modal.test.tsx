import React     from 'react'
import ReactDOM  from 'react-dom'

import { Modal } from '../src/index'

it('Modal should render correctly', () => {
  ReactDOM.createPortal = node => node

  expect(
    ReactDOM.createPortal(
      <Modal visible opacity='large' onClick={Function} onClose={Function}>
        Something
      </Modal>
    )
  ).toMatchSnapshot()
})
