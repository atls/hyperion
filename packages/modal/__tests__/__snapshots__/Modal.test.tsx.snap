// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<Modal
  onClick={[Function]}
  onClose={[Function]}
  opacity="large"
  visible={true}
>
  Something
</Modal>
`;
